cpp/src/utilities/host_memory.cpp

```diff
@@ -225,4 +282,29 @@ void set_kernel_pinned_copy_threshold(size_t threshold)
 
 size_t get_kernel_pinned_copy_threshold() { return kernel_pinned_copy_threshold(); }
 
+CUDF_EXPORT auto& allocate_host_as_pinned_threshold()
+{
+  // use pageable memory for all host allocations
+  static std::atomic<size_t> threshold = 0;
+  return threshold;
+}
+
+void set_allocate_host_as_pinned_threshold(size_t threshold)
+{
+  allocate_host_as_pinned_threshold() = threshold;
+}
+
+size_t get_allocate_host_as_pinned_threshold() { return allocate_host_as_pinned_threshold(); }
+
+namespace detail {
+
+CUDF_EXPORT rmm::host_async_resource_ref get_pageable_memory_resource()
+{
+  static new_delete_memory_resource mr{};
+  static rmm::host_async_resource_ref mr_ref{mr};
```

Comment:
question: do we need `mr_ref` to be static as well?

Reply:
we don't need it to be static, but if it's not, we'll make a new ref from the static `new_delete_memory_resource` every time we call `get_pageable_memory_resource`. I used static refs in other places, but don't think it makes a big difference either way.

Reply:
> if it's not, we'll make a new ref from the static new_delete_memory_resource every time we call get_pageable_memory_resource

That's what I was missing. Thanks!